src/pattern-library/components/patterns/data/AspectRatioPage.js

```diff
@@ -0,0 +1,171 @@
+        </p>
+      }
+    >
+      <Library.Section title="AspectRatio">
+        <Library.Pattern title="Status">
+          <p>
+            <code>AspectRatio</code> is a new simple component.
+          </p>
+        </Library.Pattern>
+
+        <Library.Pattern title="Usage">
+          <Next.Usage componentName="AspectRatio" />
+          <Library.Example>
+            <p>
+              <code>AspectRatio</code> controls the aspect ratio and containment
+              of its first direct child.
+            </p>
+            <Library.Demo
+              title="AspectRatio with default 16:9 ratio"
+              withSource
+            >
+              <div className="w-[350px]">
+                <AspectRatio>
+                  <img src="https://placekitten.com/400/400" alt="kitty" />
+                </AspectRatio>
+              </div>
+            </Library.Demo>
+          </Library.Example>
+          <Library.Example title="Placeholder content">
+            <p>
+              Placeholder content can be put in a container that is the first
+              direct child of the <code>AspectRatio</code>.
+            </p>
+            <Library.Demo title="Placeholder" withSource>
+              <div className="w-[250px] border rounded-md">
+                <AspectRatio>
+                  <div>
+                    <FileImageIcon className="text-slate-5 w-2em h-2em" />
+                  </div>
+                </AspectRatio>
+              </div>
+            </Library.Demo>
+          </Library.Example>
+          <Library.Example title="Fitting and positioning content">
+            <p>
+              The way the content is resized and fitted (
+              <Link
+                underline="always"
+                href="https://developer.mozilla.org/en-US/docs/Web/CSS/object-fit"
+              >
+                object-fit
+              </Link>
+              ) can be controlled with the <code>objectFit</code> prop
+              (documented below).
+            </p>
+            <Library.Demo title="Changing object-fit to 'contain'" withSource>
+              <div className="w-[350px] border rounded-md">
+                <AspectRatio objectFit="contain">
+                  <img src="https://placekitten.com/400/400" alt="kitty" />
+                </AspectRatio>
+              </div>
+            </Library.Demo>
+            <p>
+              <Link
+                href="https://developer.mozilla.org/en-US/docs/Web/CSS/object-position"
+                underline="always"
+              >
+                Object positioning
+              </Link>{' '}
+              within the content {"element's"} box can be handled by styling the
+              content element itself. By default, the browser will position
+              content <code>{'50% 50%'}</code> within the {"element's"} box. In
+              this example, the image is styled to apply{' '}
+              <code>{'object-position: top'}</code> instead.
+            </p>
```

Comment:
For this kind of example the behavior of the prop is best illustrated through comparison. A future improvement could add radio buttons to switch between different prop values.